nistats/design_matrix.py

```diff
@@ -446,3 +446,47 @@ def plot_design_matrix(design_matrix, rescale=True, ax=None):
     ax.set_xticks(range(len(names)))
     ax.set_xticklabels(names, rotation=60, ha='right')
     return ax
+
+
+def create_second_level_design(maps_table, confounds=None):
+    """Sets up a second level design from a maps table.
+
+    Parameters
+    ----------
+    maps_table: pandas DataFrame
+        Contains at least columns 'map_name' and 'subject_id'
+    confounds: pandas DataFrame, optional
+        If given, contains at least two columns, 'subject_id' and one confound.
+        confounds and maps_table do not need to agree on their shape,
+        information between them is matched based on the 'subject_id' column
+        that both must have.
+
+    Returns
+    -------
+    design_matrix: pandas DataFrame
+        The second level design matrix
+    """
+    maps_name = maps_table['map_name'].tolist()
+    subjects_id = maps_table['subject_id'].tolist()
+    confounds_name = []
+    if confounds is not None:
+        confounds_name = confounds.columns.tolist()
+        confounds_name.remove('subject_id')
+    design_columns = (np.unique(maps_name).tolist() +
+                      np.unique(subjects_id).tolist() +
+                      confounds_name)
+    design_matrix = pd.DataFrame(columns=design_columns)
+    for ridx, row in maps_table.iterrows():
+        design_matrix.loc[ridx] = [0] * len(design_columns)
+        design_matrix.loc[ridx, row['map_name']] = 1
+        design_matrix.loc[ridx, row['subject_id']] = 1
+        if confounds is not None:
+            conrow = confounds['subject_id'] == row['subject_id']
+            for conf_name in confounds_name:
+                design_matrix.loc[ridx, conf_name] = confounds[conrow][conf_name].values
+
+    # check column names are unique
+    if len(np.unique(design_columns)) != len(design_columns):
+        raise ValueError('Design matrix columns do not have unique names')
+
+    return design_matrix
```

Comment:
yes. At this point I would probably spit a warning rather than an Error -- it is clear that using such a design matrix will lead to aberrant estimates.